graphql/schemabuilder/pagination.go

```diff
@@ -298,23 +306,41 @@ func (o *Object) PaginateFieldFunc(name string, f interface{}) {
 		})
 }
 
-func (funcCtx *funcContext) consumePaginatedArgs(sb *schemaBuilder, in []reflect.Type) (*argParser, graphql.Type, []reflect.Type, error) {
+func embeddedPaginationArgs(argType reflect.Type) bool {
+	for i := 0; i < argType.NumField(); i++ {
+		field := argType.Field(i)
+
+		if field.Type == reflect.TypeOf(PaginationArgs{}) {
+			return true
+		}
+	}
+	return false
+}
+
+func (funcCtx *funcContext) consumePaginatedArgs(sb *schemaBuilder, in []reflect.Type) (*argParser, graphql.Type, []reflect.Type, bool, error) {
 	var argParser *argParser
 	var argType graphql.Type
 	var err error
+	embedsArgs := false
+
+	// If the args passed into paginated field func embed the PaginationArgs then the arg parser
+	// needs to be constructed differently from the default case.
 	if len(in) > 0 && in[0] != selectionSetType {
-		if argParser, argType, err = sb.buildPaginatedArgParser(in[0]); err != nil {
-			return nil, nil, nil, err
+		if embeddedPaginationArgs(in[0]) {
+			argParser, argType, err = sb.buildEmbeddedArgParser(in[0])
+			embedsArgs = true
+		} else {
+			argParser, argType, err = sb.buildPaginatedArgParser(in[0])
 		}
 		in = in[1:]
 	} else {
-		if argParser, argType, err = sb.buildPaginatedArgParser(nil); err != nil {
-			return nil, nil, in, err
-		}
+		argParser, argType, err = sb.buildPaginatedArgParser(nil)
+	}
+	if err != nil {
```

Comment:
this is kind of weird. can we keep the err handling per branch, or is there a reason to change this structuring?

Reply:
oh okay i thought it would be better since all branches return an error